packages/purpose-process/test/deletePurpose.test.ts

```diff
@@ -209,4 +367,71 @@ describe("deletePurpose", () => {
       ).rejects.toThrowError(purposeCannotBeDeleted(mockPurpose.id));
     }
   );
+  it("should throw organizationIsNotTheDelegatedConsumer when the requester is the Consumer and is deleting a purpose created by the delegate in deletePurpose", async () => {
+    const authData = getRandomAuthData();
+    const mockEService = getMockEService();
+    const mockPurposeVersion: PurposeVersion = getMockPurposeVersion(
+      purposeVersionState.draft
+    );
+    const mockPurpose: Purpose = {
+      ...getMockPurpose(),
+      eserviceId: mockEService.id,
+      versions: [mockPurposeVersion],
+      consumerId: authData.organizationId,
+      delegationId: generateId<DelegationId>(),
+    };
+
+    const delegation = getMockDelegation({
+      id: mockPurpose.delegationId,
+      kind: delegationKind.delegatedConsumer,
+      eserviceId: mockPurpose.eserviceId,
+      delegatorId: mockPurpose.consumerId,
+      delegateId: generateId<TenantId>(),
+      state: delegationState.active,
+    });
+    await addOnePurpose(mockPurpose);
+    await addOneDelegation(delegation);
+    await writeInReadmodel(toReadModelEService(mockEService), eservices);
+
+    expect(
+      purposeService.deletePurpose(mockPurpose.id, {
+        authData,
+        correlationId: generateId(),
+        logger: genericLogger,
+        serviceName: "",
+      })
+    ).rejects.toThrowError(
+      organizationIsNotTheDelegatedConsumer(
+        authData.organizationId,
+        delegation.id
+      )
+    );
+  });
+
+  it("should throw organizationIsNotTheConsumer when the requester is the Consumer with no delegation and is deleting a purpose created by a delegate in deletePurpose", async () => {
```

Comment:
Looks good!
You could also add two tests for the following **error** cases:
1. the requester is a delegate for the eservice and is deleting a purpose with no delegationId
2. the requester is a delegate for the eservice and is deleting a purpose for a different delegationId (a different delegate)